### ISSUE TITLE
Electron crash when using WebRequest OnBeforeRequest

### ISSUE THREAD

**Issue by jmeinke** [electron/electron#27603]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:**
12.0.0-beta.21
* **Operating System:**
Windows 10
* **Last Known Working Electron version:**
12.0.0-beta.20

### Expected Behavior
No crash.

### Actual Behavior
Crash reported to Sentry using minidump uploader:
https://sentry.io/share/issue/70e843318a304dae9616805c3223a1cc/

### To Reproduce this might be enough
```
Use Electron.WebRequest .onBeforeRequest(
      {
        urls: [],
      },
      ({ url }, callback) => {
        if (url.startsWith('https://test..com')) {
          callback({ cancel: true });
        } else {
          callback({ cancel: false });
        }
      },
    );
```

**Comment by vladimiry** [comment]
#27616 might potentially be related one.

**Comment by ckerr** [comment]
I think @vladimiry is correct that #27616 is a duplicate of this one. It has [another good crash report](https://github.com/electron/electron/files/5922021/dump.txt) with a slightly different crash, but both are happening inside V8 internal being called from `void ToDictionary(gin::Dictionary* details, extensions::WebRequestInfo* info)` in `shell/browser/api/electron_api_web_request.cc`

In addition, 27616 was reported against Linux so I'm going to change the `platform/` labels to note this isn't platform-dependent.

**Comment by ckerr** [comment]
Looks like a big part of that `ToDictionary()` function was changed 25 days ago with https://github.com/electron/electron/commit/088f2e625f475056515e6000885ff404a02b2fdc for https://github.com/electron/electron/pull/27078

**Comment by VerteDinde** [comment]
@jmeinke @vladimiry Out of curiosity - are you both also seeing in this behavior of 12.0.0-beta.22 or later? We merged a fix to a similar crash on loading extensions with WebRequest, and I'm trying to rule out if that might affect this as well.

**Comment by vladimiry** [comment]
I've been running https://github.com/vladimiry/ElectronMail assembled on 12.0.0-beta.24 for a few hours now and it works nicely, no crashes. So can confirm that the original issue appears to be resolved for my case somewhere between 12.0.0-beta.22 and 12.0.0-beta.24, but I still keep testing it. I'm even thinking of releasing a new app release on 12.0.0-beta.24.

Another not directly related note regarding v12-betas. When I was trying early v12-beta versions the UI built on Angular was getting complete frozen/unresponsive after like 10-20 minutes of running the app. But there is no such issue anymore in newest beta versions. So the issue got recently resolved somewhere in @electron / @chromium / @angular.

**Comment by VerteDinde** [comment]
Given that this seems to be fixed in the newer beta releases, I'm going to close this out.
